int16 Bottom0
int16 Top0
int16 Bottom1
int16 Top1
